ps_facetedsearch.php

```diff
@@ -1438,27 +1438,27 @@ public function getProductByFilters(
         Db::getInstance()->execute('DROP TEMPORARY TABLE IF EXISTS '._DB_PREFIX_.'cat_filter_restriction', false);
         if (empty($selected_filters['category'])) {
             /* Create the table which contains all the id_product in a cat or a tree */
-            Db::getInstance()->execute('CREATE TEMPORARY TABLE '._DB_PREFIX_.'cat_filter_restriction ENGINE=MEMORY
-                                                        SELECT cp.id_product, MIN(cp.position) position FROM '._DB_PREFIX_.'category c
-                                                        STRAIGHT_JOIN '._DB_PREFIX_.'category_product cp ON (c.id_category = cp.id_category AND
+            Db::getInstance()->execute('CREATE TEMPORARY TABLE '._DB_PREFIX_.'cat_filter_restriction ENGINE=MEMORY 
+                                                        SELECT cp.id_product, cp.position FROM '._DB_PREFIX_.'category_product cp 
+                                                        STRAIGHT_JOIN '._DB_PREFIX_.'category c ON (c.id_category = cp.id_category AND
                                                         '.($this->ps_layered_full_tree ? 'c.nleft >= '.(int) $parent->nleft.'
                                                         AND c.nright <= '.(int) $parent->nright : 'c.id_category = '.(int) $id_parent).'
                                                         AND c.active = 1)
                                                         STRAIGHT_JOIN `'._DB_PREFIX_.'product` p ON (p.id_product=cp.id_product)
                                                         '.$price_filter_query_in.'
                                                         '.$query_filters_from.'
                                                         WHERE 1 '.$query_filters_where.'
-                                                        GROUP BY cp.id_product ORDER BY position, id_product', false);
+                                                        GROUP BY cp.id_product ORDER BY cp.position, cp.id_product', false);
```

Comment:
with the GROUP BY, if you do not use MIN(cp.position) you don't know which position you will get if there are several time the same id_product in different cat, but with different position. The current code makes sure you always use the smallest position from all possible cats, but with your code, your using a random position.